services/src/main/java/com/linkedin/kmf/services/ProduceService.java

```diff
@@ -86,6 +88,14 @@ public ProduceService(Map<String, Object> props, String name) throws Exception {
     _partitionNum = new AtomicInteger(0);
     _running = new AtomicBoolean(false);
     _nextIndexPerPartition = new ConcurrentHashMap<>();
+    _producerPropsOverride = props.containsKey(ProduceServiceConfig.PRODUCER_PROPS_CONFIG) ?
+      (Map) props.get(ProduceServiceConfig.PRODUCER_PROPS_CONFIG) : new HashMap<>();
+
+    for (String property: NONOVERRIDABLE_PROPERTIES) {
+      if (_producerPropsOverride.containsKey(property)) {
+        throw new ConfigException("Override must not contain bootstrap or zookeeper config.");
```

Comment:
Should this message be `"Property " + property " should not be specified inside " + ProduceServiceConfig.PRODUCER_PROPS_CONFIG`?

Same for the error message in ConsumeService